addons/mail/wizard/mail_compose_message.py

```diff
@@ -241,7 +241,7 @@ def _gc_lost_attachments(self):
 
     def action_send_mail(self):
         """ Used for action button that do not accept arguments. """
-        self._action_send_mail(auto_commit=False)
+        self.with_context(mailing_document_based=True)._action_send_mail(auto_commit=False)
```

Comment:
This means we'll never check for duplicates though :/
No good

Reply:
I've checked and yes there is a ``mail.compose.message`` model in ``account`` but when I add ``action_send_mail()`` to it, it's never trigged.

Reply:
the account.invoice.send model (if it's what we are talking about) already seems to have mail_document_based in the python code

this scenario is about "Send receipts by email" that is using mail.compose.message directly:

https://github.com/odoo/odoo/blob/ddc9fe0efee0df25d6e4cc8e33c859f500fe8ec1/addons/account/views/account_payment_view.xml#L446-L460

So I guess we should for example have the mailing_document_based context in that action.

Reply:
I found that we missed something. ``account.invoice.send`` doesn't extend the ``mail.compose.message`` model but creates a new one with the fields of ``mail.compose.message``
```python
class AccountInvoiceSend(models.TransientModel):
    _name = 'account.invoice.send'
    _inherits = {'mail.compose.message':'composer_id'}
```
That's why it was never trigger when I added ``def action_send_mail()`` to it.

I had to create a proper extension of it to add the context to ``action_send_mail()``